pkg/config/tmpl/dottedconfig_test.go

```diff
@@ -50,3 +50,54 @@ func TestMakeDottedConfig(t *testing.T) {
 		})
 	}
 }
+
+func clone(m map[string]any) map[string]any {
```

Comment:
Could we use [maps.Clone](https://pkg.go.dev/maps@master#Clone) instead?

Reply:
No, because we need a deep clone, not a shallow one.